agent/command/git.go

```diff
@@ -219,8 +219,14 @@ func (opts cloneOpts) buildHTTPCloneCommand(forApp bool) ([]string, error) {
 	}
 	var gitURL string
 	if forApp {
+		if opts.token == "" {
```

Comment:
Isn't the token already checked during [options validation](https://github.com/evergreen-ci/evergreen/blob/839e7fb0154aa9a9280615594b51440ad1b70bc0/agent/command/git.go#L123-L125)?